notebook/static/notebook/js/menubar.js

```diff
@@ -128,6 +128,133 @@ define([
 
         this._new_window(url);
     };
+    
+    MenuBar.prototype._nbconvert_upload_conf = function (download) {
+      
+      var body =  $('<div>');
+      var notebook_path = utils.encode_uri_components(this.notebook.notebook_path);
+
+      var create_json = function(notebook, config){
+        var json_to_pass = {
+          notebook: notebook.toJSON(),
+          config: config
+        };
+        return json_to_pass;
+      };
+
+      
+      var form  = $('<form>');
+      var fileinput = $('<input>')
+        .attr('type', 'file')
+        .attr('tabindex', '1');
+      var fileformat = $('<select>');
+      fileformat.append(
+        $('<option>markdown</option>')
+          .attr('value','markdown')
+      );
+      fileformat.append(
+        $('<option/>')
+          .attr('value','html')
+          .text('html')
+      );
+      fileformat.append(
+        $('<option>rst</option>')
+          .attr('value','rst')
+      );
+      fileformat.append(
+        $('<option>pdf</option>')
+          .attr('value','pdf')
+      );
+      fileformat.append(
+        $('<option>latex</option>')
+          .attr('value','latex')
+      );
+      fileformat.append(
+        $('<option>script</option>')
+          .attr('value','script')
+      );
+      
+      form.append(fileformat);
+      form.append(fileinput);
+      body.append(form);
+      
+      var notebook_path = utils.encode_uri_components(this.notebook.notebook_path);
+
+      var that = this;
+
+      var handle_nbconvert_post = function(){
+        var filereader = new FileReader();
+        filereader.onload = function(){
+          var my_config_data = filereader.result;
+          that['json_content'] = create_json(that.notebook, my_config_data);
+          on_done();
+        };
+        if (fileinput[0].files.length>0){
+          filereader.readAsText(fileinput[0].files[0]);
+        }
+        else{
+          that['json_content']=create_json(that.notebook, '{}')
+          on_done();
+        }
+      };
+      var on_done = function(){
+        var url = utils.url_path_join(
+            that.base_url,
+            'nbconvert',
+            fileformat.val(),
+            notebook_path
+        ) + "?download=" + download.toString();
+          
+        var xsrf_token = utils._get_cookie('_xsrf');
+        var xhr = new XMLHttpRequest();
+        xhr.open('POST', url, true);
+        xhr.responseType = 'blob';
+        xhr.withCredentials = true;
+        xhr.setRequestHeader('X-XSRFToken', xsrf_token);
+        xhr.onreadystatechange = function(){
+          if(xhr.readyState === 4){
+            var blob = xhr.response;
```

Comment:
I don't know exactly what the issue is, but I hope we can work out how to do this with a more familiar API than the low-level XMLHttpRequest interface.